### DISCUSSION TITLE
Can you customize stats option to hide stuff like asset names, asset file sizes, "1 related asset" etc?

### DISCUSSION THREAD

**Discussion by Jakobud** [webpack/webpack#13111] [Q&A]
Some example `stats` output could be this:

```
[webpack-cli] Compilation finished
asset bundle.js 72.6 KiB [emitted] [minimized] (name: main) 1 related asset
```

Is it possible to customize this to hide certain things like:

* the word "asset" at the beginning
* the file size
* "1 related asset"
* the asset "name" (example here its (name: main)

**Comment by alexander-akait** [comment]
Yes, we have hooks for printing, you can look here https://github.com/webpack/webpack/blob/master/lib/stats/DefaultStatsPrinterPlugin.js

**Reply by Jakobud** [reply]
Oh interesting okay thanks. Is there documentation on this?

**Reply by alexander-akait** [reply]
No good docs, but you can help us to improve this, for example put in docs simple examples

**Reply by Jakobud** [reply]
I've been looking at this but drawing a blank. Not sure where to start. Is DefaultStatsPrinterPlugin something I need to import into my webpack config before using it? What are the hooks?

**Reply by sokra** [reply]
Here is an example snippet you can put into your config: (It hides the `1 related assets`)

``` js
plugins: [
  compiler => {
    compiler.hooks.compilation.tap("webpack.config.js", compilation => {
      compilation.hooks.statsPrinter.tap("webpack.config.js", statsPrinter => {
        statsPrinter.hooks.print.for("asset.filteredRelated").tap("webpack.config.js", (filteredRelated) => "");
      });
    });
  }
]
```

Swap `asset.filteredRelated` with other things to override the printing of other parts of the stats.

Here are a few interesting ones:

* `asset.type` is the `asset` work in the beginning.
* `asset.size` is the size of the asset.
* `asset.name` is the name of the asset.
* `compilation.assets` If you like to take full control over the assets rendering.

Note that there is a second argument for the `print` hook which has context information, helpers and colors.

If you add

``` js
/** @type {import("webpack").Configuration} */
module.exports = {
  ...
```

to your configuration you will have typescript typings and completion.
